practice_points.c

```diff
@@ -0,0 +1,107 @@
+#include <cstring>
+#include <memory>
```

Comment:
Smells like c++, should change filename to `practice_points.cpp` for it to compile.